server/services/lostBedService.ts

```diff
@@ -17,12 +18,59 @@ export default class LostBedService {
     return confirmedLostBed
   }
 
-  async getLostBed(token: string, premisesId: string, id: string): Promise<LostBed> {
+  async getLostBed(token: string, premisesId: string, id: string): Promise<LostBedSummary> {
     const lostBedClient = this.lostBedClientFactory(token)
+    const premisesClient = this.premisesClientFactory(token)
 
     const lostBed = await lostBedClient.find(premisesId, id)
+    const bed = await premisesClient.getBed(premisesId, lostBed.bedId)
 
-    return lostBed
+    return { ...lostBed, bedName: bed.name, roomName: bed.roomName }
+  }
+
+  async updateLostBed(
+    token: string,
+    id: string,
+    premisesId: string,
+    updateData: { endDate: string; notes: string },
+  ): Promise<LostBed> {
+    const lostBedClient = this.lostBedClientFactory(token)
+
+    const lostBed = await lostBedClient.find(premisesId, id)
+
+    // TODO: utility function to take care of creating updated data
+    lostBed.endDate = updateData.endDate
+    lostBed.notes = updateData.notes
+
+    const lostBedData = {
+      startDate: lostBed.startDate,
+      endDate: updateData.endDate,
+      reason: lostBed.reason.id,
+      referenceNumber: lostBed.referenceNumber,
+      notes: updateData.notes,
+    }
+
+    const updatedLostBed = await lostBedClient.update(id, lostBedData, premisesId)
+
+    return updatedLostBed
+  }
+
+  async getLostBeds(token: string, premisesId: string): Promise<Array<LostBedSummary>> {
+    const lostBedClient = this.lostBedClientFactory(token)
+    const lostBeds = await lostBedClient.getLostBeds(premisesId)
+    const premisesClient = this.premisesClientFactory(token)
+    const lostBedSummarys = await Promise.all(
+      lostBeds.map(async lostBed => {
+        const bed = await premisesClient.getBed(premisesId, lostBed.bedId)
+        return {
```

Comment:
we need to show the bed and room names on frontend, is creating a join type like this the best way forward?

Reply:
I think if we need this stuff, it'd be better if we update the API, rather than making a ton of extra API calls. I'll sort that out 👍